public_html/wp-content/plugins/pattern-creator/src/store/actions.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { dispatch, select } from '@wordpress/data-controls';
+
+/**
+ * Internal dependencies
+ */
+import { KIND, MODULE_KEY, POST_TYPE } from './utils';
+
+export function editBlockPattern( patternId ) {
+	return { type: 'EDIT_BLOCK_PATTERN', value: patternId };
+}
+
+/**
+ * Save a block pattern.
+ */
+export function* saveBlockPattern() {
+	const patternId = yield select( MODULE_KEY, 'getEditingBlockPatternId' );
+
+	// @todo maybe check for errors?
+	yield dispatch( 'core', 'saveEditedEntityRecord', KIND, POST_TYPE, patternId );
```

Comment:
The result from `select` isn't a promise, so yield and await are not interchangeable. We're not (only) waiting for a response, we want to generate multiple actions from this "action creator". So in the process of this function, we fire off the select action, which resolves into the patternId, then we keep going to fire off the save action. It's a bit overcomplicated for the current functionality, but we'll want to create other actions later (in case of errors, to save other meta, etc).

Reply:
Ah, ok.